baseline/client/tests/dashboard.test.js

```diff
@@ -279,22 +280,61 @@ describe("dashboard", () => {
-        // check Fluttershy's EEG T1 checkbox
         const fluttershyRow = document.querySelector(`[data-user-id="${fluttershyId}"]`);
-        const [_, __, dailyTasksT1Cell, ...___] = fluttershyRow.querySelectorAll("td");
-        dailyTasksT1Cell.querySelector("input").dispatchEvent(new MouseEvent("click", {bubbles: true}));
-        // wait for setTimeout in click event handler to resolve
-        await new Promise((resolve) => setTimeout(resolve));
+        const [_, __, eegT1Cell, ...___] = fluttershyRow.querySelectorAll("td");
+        expect(mc.users.get(fluttershyId).progress?.eegT1).toBeFalsy();
+        expect(eegT1Cell.querySelector("input").checked).toBe(false);
+        expect(eegT1Cell.querySelector("span").textContent).toBeFalsy();
+        // check Fluttershy's EEG T1 checkbox
+        eegT1Cell.querySelector("input").dispatchEvent(new MouseEvent("click", {bubbles: true}));
+        // wait for the async click event handler to resolve
+        await new Promise(process.nextTick);
         // Fluttershy should now have a timestamp for EEG T1
-        expect(dailyTasksT1Cell.querySelector("input").checked).toBe(true);
-        expect(Boolean(mc.users.get(fluttershyId).progress?.eegT1)).toBe(true);
+        expect(mc.users.get(fluttershyId).progress?.eegT1).toBeTruthy();  // backend is updated
+        expect(eegT1Cell.querySelector("input").checked).toBe(true);  // checkbox is checked
+        expect(eegT1Cell.querySelector("span").textContent).toBe("2010-10-10");  // timestamp is displayed
+        expect(dateSpy).toHaveBeenCalledTimes(1);
+        // restore mocks
+        dateSpy.mockRestore();
+    });
+
+    it("updates backend through client on uncheck", async () => {
+        // mock window.confirm
+        const confirmSpy = jest.spyOn(window, "confirm").mockImplementation(() => true);
+        // create dashboard
+        const mc = new MockClient(users, results);
+        const {wrapper, error, table, details} = getDashboardElements();
+        const dash = new Dashboard(table, mc);
+        await dash.refreshRecords();
+        dash.showActive();
+        // Twilight Sparkle should have a timestamp for MRI T1
+        const twiId = "95240257-42f9-4ae6-b989-0126f595e547";
+        const twiRow = document.querySelector(`[data-user-id="${twiId}"]`);
+        const [_, __, ___, mriT1Cell, ...____] = twiRow.querySelectorAll("td");
+        expect(mc.users.get(twiId).progress?.mriT1).toBeTruthy();
+        expect(mriT1Cell.querySelector("input").checked).toBe(true);
+        expect(mriT1Cell.querySelector("span").textContent).toBeTruthy();
+        // uncheck Twilight Sparkle's MRI T1 checkbox
+        mriT1Cell.querySelector("input").dispatchEvent(new MouseEvent("click", {bubbles: true}));
+        // wait for the async click event handler to resolve
+        await new Promise(process.nextTick);
+        // Twilight Sparkle should now have a timestamp for MRI T1
```

Comment:
I think "now" is probably supposed to be "not" here.

Reply:
Nice catch! Just fixed it.